modules/new_profile/jsx/NewProfileIndex.js

```diff
@@ -0,0 +1,260 @@
+import Panel from 'Panel';
+import Loader from 'Loader';
+/**
+ * New Profile Form
+ *
+ * Create a new profile form
+ *
+ * @author  Shen Wang
+ * @version 1.0.0
+ * */
+class NewProfileIndex extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: {},
+      configData: {},
+      formData: {},
+      newData: {},
+      isLoaded: false,
+      isCreated: false,
+      errMessage: '',
+    };
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.setFormData = this.setFormData.bind(this);
+    this.fetchData = this.fetchData.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchData()
+      .then(() => this.setState({isLoaded: true}));
+  }
+
+  /**
+   * Retrieve data from the provided URL and save it in state
+   *
+   *
+   * @return {object}
+   */
+  fetchData() {
+    return fetch(this.props.dataURL,
+      {credentials: 'same-origin'})
+      .then((resp) => resp.json())
+      .then((data) => this.setState({configData: data}))
+      .catch((error) => {
+        this.setState({error: true});
+      });
+  }
+  /**
+   * It will check the date of birth and DEC, the date fileds must match.
+   * If there is a error, it will return true.
```

Comment:
That's a really confusing convention.. usually validate functions return true on success and false on error.